src/dmd/dcast.d

```diff
@@ -44,11 +44,22 @@ import dmd.visitor;
 
 enum LOG = false;
 
-/**************************************
- * Do an implicit cast.
- * Issue error if it can't be done.
+/**
+ * Attempt to implicitly cast the `from` expression into `target` type.
+ *
+ * This routine will change `from`. To check the matching level,
+ * use `implicitConvTo`.
+ *
+ * Params:
+ *   from = Expression that is to be casted
+ *   sc = Current scope
+ *   target = Expected resulting type
+ *
+ * Returns:
+ *   The resulting casted expression (mutating `from`), or `ErrorExp`
+ *    if such an implicit conversion is not possible.
  */
-Expression implicitCastTo(Expression e, Scope* sc, Type t)
+Expression implicitCastTo(Expression from, Scope* sc, Type target)
```

Comment:
DMD consistently uses `e` for an Expression and `t` for a type. It is not an improvement to come up with new naming style for them.

Reply:
Yes it uses `e` and then `e1` and `e2` and so on. That's a good example of bad Hungarian notation, as the name does not give any info on its intent. I'm not saying "from" and "target" is the best name out there, but it at least gives more information to the reader as to what the parameter is for. 

If the issue is solely about consistency, I could do a large amount of renames in a single PR, but those changes are usually frowned upon (for good reason IMO), and I recall being asked to make those more gradual, which this is.

Reply:
I've been dealing with DMD for a very long time and it is very useful to know that `e` is an expression and `t` is a type. It's used everywhere in DMD. Please do not start on renaming all of this, gradual or not. I can go further and assert that `from` is hopelessly generic, as all `in` parameters are "from". `target` is normally used in DMD to denote the code generation target, besides not actually being a target since the result is not written to it.

Pretty much everyone has their own ideas on how things should be named, there is no "correct" way that will please more than one person. But the incorrect way is to have lots of people with their own unique stylistic ideas each refactoring bits of the code, as then it just becomes incoherent.

Looking at the code to make it flow better, improving Ddoc comments, doing single assignment, minimizing variable lifetimes, adding const where possible, replacing pointers with ref and out, putting public functions at the top, are all likely worthwhile refactorings.

P.S. I know that Hungarian notation got a bad rap from the way Microsoft used it in Windows, but that doesn't mean it's all bad. Just like OOP has its uses, so does HN.

Reply:
>`target` is normally used in DMD to denote the code generation target, besides not actually being a target since the result is not written to it.

Pretty much the only takeaway from this renaming, you run the risk of conflicting with already existing globals.

Reply:
> P.S. I know that Hungarian notation got a bad rap from the way Microsoft used it in Windows, but that doesn't mean it's all bad. Just like OOP has its uses, so does HN.

And I totally agree, hence why I mentioned "bad" Hungarian notation. Having a syntax to denote the categorization of a variable is good, although less useful nowadays (IIRC, the story was that it caught on at Microsoft after it was used in Excel successfully, notably to in their grid system, as an `c` would be used for `column` and `r` for `row`).

The bad rap it got was exactly because people started to add the type to the notation, such as `p` for pointer, which we do in many places in DMD. I don't mind Hungarian notation, I mind variable names without a meaning. However, if you feel that there's no "right" way to name those, and we should stick with `e`, I'll stick with `e`.

I had no intention to go through the code base and rename them. But if I were to suggest a single improvement in the spirit of adding more meaning to variable names, from the top of my head, `BinExp.{e1,e2}` => `BinExp.{lhs,rhs}` would be one.

Reply:
> Pretty much the only takeaway from this renaming, you run the risk of conflicting with already existing globals.

All `visit` methods within that function take an `Expression` or derivative as an argument, and its name is `e`. Before worrying about globals, we should worry about locals.